### PR TITLE
fix: fix unexpected parser

### PR DESCRIPTION
**Pull request by Zxilly**
FIX: #240 

Signed-off-by: Zxilly <zhouxinyu1001@gmail.com>

### CODE REVIEW

**Comment by Zxilly** [comment]
I think empty line with `/` should be regarded as an error, but not try to parse it.

File: test/config/testini.ini [hunk 1]
```diff
@@ -43,5 +43,10 @@ name = \
 
 [multi5]
 name = r.sub==p.sub \
-   &&r.obj==p.obj\
+   &&r.obj==p.obj \
    \
+
+[multi6]
```
Comment (hsluoyz) [inline test/config/testini.ini:49]: We need a new example like:

```ini
[matchers]
m = g(r.sub, p.sub, r.cid+'::'+r.lid) \
&& r.method == p.method \
&& keyMatch(r.cid, p.cid) \
&& keyMatch(r.lid, p.lid) \
&& keyMatch2(r.obj, p.obj) 
```

The key point here is that there should be one line that contains no "="

The `[multi6]` example doesn't seem to show such a corner case.

File: src/config.ts [hunk 2]
```diff
@@ -91,7 +91,10 @@ export class Config implements ConfigInterface {
   }
 
   private parseBuffer(buf: Buffer): void {
-    const lines = buf.toString().split('\n');
+    const lines = buf
+      .toString()
+      .split('\n')
+      .filter((v) => v);
```
Comment (hsluoyz) [inline src/config.ts:97]: Why separate to 4 lines? The original row length is not exceeding the limit I think.

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Comment by hsluoyz** [comment]
@hariprasadiit please review.

File: src/config.ts (hunk 2 above)
Comment (Zxilly) [reply src/config.ts:97]: this was fromatted by prettier,eslint forced this format

File: test/config/testini.ini (hunk 1 above)
Comment (Zxilly) [reply test/config/testini.ini:49]: In fact,the problem was caused by the empty line at the end of file,but not the `=` existed or not

File: src/config.ts (hunk 2 above)
Comment (hsluoyz) [reply src/config.ts:97]: OK

File: test/config/testini.ini (hunk 1 above)
Comment (hsluoyz) [reply test/config/testini.ini:49]: @Zxilly Oh, this is really a strange bug then. Nice catch.

@hariprasadiit is this the case?

Comment (hariprasadiit) [reply test/config/testini.ini:49]: @hsluoyz yes, seems it's empty line issue. If we don't have empty line at the end, error is thrown.

@Zxilly Nice catch 👍

**Comment by hariprasadiit** [comment]
This PR fixes new line issue. 

Thanks

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**